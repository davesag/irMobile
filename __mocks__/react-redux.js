const reactRedux = jest.genMockFromModule('react-redux')

reactRedux.connect.mockImplementation(() => fn => fn)

module.exports = reactRedux
