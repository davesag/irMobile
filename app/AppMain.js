/**
 * @format
 */
import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { createAppContainer, createStackNavigator } from 'react-navigation'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
// import Config from 'react-native-config'
import { View } from 'react-native'

import { restoreKeys } from './services/ir/actions'

import SplashScreen from './screens/Splash'
import splashNav from './screens/Splash/navigationOptions'

// Establish stack navigator
const StackNavigator = createStackNavigator(
  {
    Splash: {
      screen: SplashScreen,
      navigationOptions: splashNav
    }
  },
  { mode: 'modal' }
)

const AppContainer = createAppContainer(StackNavigator)

// TODO: Add https://github.com/joinspontaneous/react-native-loading-spinner-overlay
class AppMain extends PureComponent {
  static propTypes = {
    doRestoreKeys: PropTypes.func.isRequired
  }

  componentDidMount() {
    this.props.doRestoreKeys()
  }

  render() {
    return (
      <View style={{ flex: 1 }}>
        <AppContainer />
      </View>
    )
  }
}

export const mapDispatchToProps = dispatch =>
  bindActionCreators({ doRestoreKeys: restoreKeys }, dispatch)

export default connect(
  null,
  mapDispatchToProps
)(AppMain)
