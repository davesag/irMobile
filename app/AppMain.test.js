import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import * as reactRedux from 'react-redux'

import AppMain, { mapDispatchToProps } from './AppMain'

let tree

describe('mapDispatchToProps', () => {
  const dispatchProps = mapDispatchToProps(jest.fn())

  it('returns bound function to dispatch', () => {
    expect(dispatchProps).toEqual({
      doRestoreKeys: expect.any(Function)
    })
  })
})

describe('rendering the component', () => {
  const doRestoreKeys = jest.fn()

  beforeAll(() => {
    tree = renderer.create(<AppMain doRestoreKeys={doRestoreKeys} />)
    tree.getInstance().componentDidMount()
  })

  it('rendered correctly ', () => {
    expect(tree).toMatchSnapshot()
  })

  it('connected with redux', () => {
    expect(reactRedux.connect).toHaveBeenCalled()
  })

  it('called doRestoreKeys', () => {
    expect(doRestoreKeys).toHaveBeenCalled()
  })
})
