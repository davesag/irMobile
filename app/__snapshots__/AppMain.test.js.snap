// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering the component rendered correctly  1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
